### PR TITLE
doc: add more details about QEMU support

### PR DESCRIPTION
**Pull request by ppannuto**
### Pull Request Overview

This pull request adds some documentation and caveats to QEMU support.

### Testing Strategy

N/A

### TODO or Help Wanted

Probably still more documentation, but this is at least a step in the right direction and hopefully tempers expectations a bit.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: doc/Qemu.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+<!-- tocstop -->
+
+## Supported Boards
+
+QEMU support for embedded platforms is limited. Please check the the table in
+the [`boards/` subdirectory](../boards/README.md) for an up-to-date list of
+supported boards.
+
+## Tock QEMU Fork
```
Comment (alistair23) [inline doc/Qemu.md:26]: I'm preparing the PR now, so the fork won't be required in 2/3 days

**Comment by alistair23** [comment]
Looks good, will need to be updated once everything is upstream. Is it maybe worth holding off a few days and then not mention the fork?

**Comment by ppannuto** [comment]
Sure! That's exciting news, I didn't realize upstreaming was so imminent!

**Comment by ppannuto** [comment]
Rolled up into #1971.

**Pull request closed without merging**